inc/views/pluggable/pagination.php

```diff
@@ -22,12 +22,62 @@ class Pagination extends Base_View {
 	 * @return void
 	 */
 	public function init() {
+		add_action( 'rest_api_init', array( $this, 'register_endpoints' ) );
 		add_filter( 'neve_filter_main_script_localization', array( $this, 'filter_localization' ) );
 		add_action( 'neve_do_pagination', array( $this, 'render_pagination' ) );
 		add_action( 'neve_post_navigation', array( $this, 'render_post_navigation' ) );
+	}
+
+	/**
+	 * Register Rest API endpoint for posts retrieval.
+	 */
+	public function register_endpoints() {
+		register_rest_route(
+			'posts/v1',
+			'/page/(?P<page_number>\d+)/',
+			array(
+				'methods'  => \WP_REST_Server::READABLE,
+				'callback' => array( $this, 'get_posts' ),
+			)
```

Comment:
we can look in the future to re-use the core routes and load the content thru an underscore template as the current approach is not ideal